internal/core/core.go

```diff
@@ -99,6 +99,19 @@ type Array struct {
 	Shape []int `json:"shape" swaggertype:"array,integer" example:"10,50"`
 }
 
+// @Description Slice bounds.
+type Bound struct {
+	// Direction of the bound. See SliceRequest.Direction for valid options
+	Direction string  `json:"direction" binding:"required" example:"inline"`
+
+	// Lower bound - inclusive
+	Lower float32 `json:"lower" binding:"required" example:"100"`
+
+	// Upper bound - inclusive
+	// Upper bound must be greater or equal to lower bound
+	Upper float32 `json:"upper" binding:"required" example:"200"`
```

Comment:
The struct parsing (from json) will fail if they are not supplied. But yes, you can still make invalid Bounds objects e.g. in tests. But in production code these are only every created when parsing the json request

Reply:
I am not sure what you mean because I pretty much tested that before writing a comment.
And I still confirm that 

```
"bounds": [
    {"direction": "i", "upper": 2}
],

```
succeeds

```
"bounds": [
    {"direction": "inline", "upper": 5}
],
```

fails with invalid range.
```

"bounds": [
    { "lower": 3, "upper": 5}
],
```

fails with invalid direction

And this is not a "required" problem, this is a default value problem. Reason why we supply ints as *int.